File: ios/YATATests/TodoListTests.swift
//
//  TodoListTests.swift
//  YATATests
//
//  Created by Alpár Szotyori on 20.05.24.
//

import XCTest
@testable import YATA

final class TodoListTests: XCTestCase {

    override func setUpWithError() throws {
        // Put setup code here. This method is called before the invocation of each test method in the class.
    }

    override func tearDownWithError() throws {
        // Put teardown code here. This method is called after the invocation of each test method in the class.
    }

    func testNewTodoListHasRandomUniqueId() throws {
        // Given
        let todoList1 = TodoList()
        let todoList2 = TodoList()

        // Then
        XCTAssertTrue(todoList1.id != todoList2.id)
    }

    func testNewTodoListHasNoTodos() {
        // Given
        let todoList = TodoList()

        // Then
        XCTAssertTrue(todoList.todos.isEmpty)
    }

}

File: android/app/src/test/java/de/repeatuntil/yata/domain/TodoListTests.kt
package de.repeatuntil.yata.domain

import de.repeatuntil.yata.domain.todolist.TodoList
import de.repeatuntil.yata.domain.user.User
import org.junit.Assert
import org.junit.Test

class TodoListTests {

    @Test
    fun `new to-do list has a random unique id`() {
        // Given
        val todoList1 = TodoList()
        val todoList2 = TodoList()

        // Then
        assert(todoList1.id != todoList2.id)
    }

    @Test
    fun `new to-do list has no user`() {
        // Given
        val todoList = TodoList()

        // Then
        Assert.assertNull(todoList.userId)
    }

    @Test
    fun `new to-do list has no to-dos`() {
        // Given
        val todoList = TodoList()

        // Then
        assert(todoList.todos.isEmpty())
    }
}